src/librustc/infer/mod.rs

```diff
@@ -1555,30 +1637,33 @@ impl<'a, 'gcx, 'tcx> InferCtxt<'a, 'gcx, 'tcx> {
                         def_id: DefId)
                         -> Option<ty::ClosureKind>
     {
-        if def_id.is_local() {
-            self.tables.borrow().closure_kinds.get(&def_id).cloned()
-        } else {
-            // During typeck, ALL closures are local. But afterwards,
-            // during trans, we see closure ids from other traits.
-            // That may require loading the closure data out of the
-            // cstore.
-            Some(self.tcx.closure_kind(def_id))
+        if let InferTables::Fresh(tables) = self.tables {
+            if let Some(id) = self.tcx.map.as_local_node_id(def_id) {
```

Comment:
"fresh" is a bad name because it means "currently inferred" (in typeck). The short answer is "everywhere outside typeck".